### PR TITLE
[test] Fori loop simple case without hard-code

### PR DESCRIPTION
**Pull request by ManfeiBai**
code based on https://github.com/pytorch/xla/pull/7012, create post-order tracing interface on python level, to enable post-order tracing for XLAComputation like https://github.com/pytorch/xla/blob/master/torch_xla/csrc/xla_graph_executor.cpp#L1293

### CODE REVIEW

File: torch_xla/csrc/xla_graph_executor.h [hunk 1]
```diff
@@ -187,6 +187,14 @@ class XLAGraphExecutor : public torch::lazy::LazyGraphExecutor {
   void ClearPendingIrs(std::vector<XLATensorPtr> tensors,
                        const torch::lazy::BackendDevice& device);
 
+//   struct saveComputation {
+//     runtime::ComputationClient::ComputationPtr computation;
+//   };
+//   std::vector<runtime::ComputationClient::ComputationPtr>
+  runtime::ComputationClient::ComputationPtr GetXLAComputation(
```
Comment (ManfeiBai) [inline torch_xla/csrc/xla_graph_executor.h:194]: Hi, @miladm, this is the script code for post-order tracing compile function

File: torch_xla/csrc/xla_graph_executor.cpp [hunk 2]
```diff
@@ -1514,4 +1517,170 @@ XLAGraphExecutor::SyncTensorsGraphInternal(
...
+
+  // CompilationResult compile_result =
+  //     Compile(*tensors, devices, coll, &po_data, ir_values);
+
+  // runtime::ComputationClient::ComputationPtr 
+  // saveComputation* compile_result = std::move(
+  //   Compile(*tensors, devices, coll, &po_data, ir_values).computation)
+  XLAGraphExecutor::saveComputation* compile_result = Compile(*tensors, devices, coll, &po_data, ir_values).computation
+  return compile_result
+
```
Comment (ManfeiBai) [inline torch_xla/csrc/xla_graph_executor.cpp:1584]: Hi, @miladm, this is the script code implementation for post-order tracing compile function

**Comment by JackCaoG** [comment]
I have 
```
  def test_while_loop_tpu_simple_linear(self):

    xm.mark_step()
    device = xm.xla_device()
    #device = ''
    torch.set_grad_enabled(False)

    class SimpleWithLinear(torch.nn.Module):
      def __init__(self):
          super().__init__()
          self.linear = torch.nn.Linear(2, 2)
          self.register_buffer("dec", torch.tensor(1))

      def forward(self, iter, x):
          def cond_fn(it, x):
              return it - self.dec > 0

          def body_fn(it, x):
              return it - 1, self.linear(x)

          return while_loop(cond_fn, body_fn, (iter, x))
      
    simple_with_linear = SimpleWithLinear()
    simple_with_linear.to(device)
    #breakpoint()
    input = torch.randn(2, 2).to(device)
    iter = torch.tensor(3, device=device)
    res = simple_with_linear(iter, input)
```
which works on CPU device but error out for XLA devices. Let's try to get this simple example working.

File: test/test_test_fori_loop.py
Comment (ManfeiBai) [inline test/test_test_fori_loop.py:1]: script used for benchmarking

**Comment by ManfeiBai** [comment]
thanks, based on this PR, created and updated to simple test case in https://github.com/pytorch/xla/pull/7094, would close this pr for now

**Pull request closed without merging**